iothub/service/src/Messaging/MessageFeedbackProcessorClient.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> for handling cloud-to-device message feedback.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-c2d#message-feedback"/>.
+    public class MessageFeedbackProcessorClient : IDisposable
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly IotHubConnection _connection;
```

Comment:
You'll need to expose a callback in this class that notifies users when the connection is lost. Otherwise, they may successfully call open, wait for feedback, and then never get it because the connection was dropped. See [here](https://github.com/Azure/azure-iot-sdk-java/blob/c35f7782e400d3d7cc341b370117e70ece76236f/service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/messaging/MessageFeedbackProcessorClient.java#L43) for the Java equivalent

Reply:
All these 3 AMQP subclients will need this feature since all 3 rely on a persistent connection.